kpatch-build/create-diff-object.c

```diff
@@ -760,11 +760,72 @@ static void kpatch_compare_symbols(struct list_head *symlist)
 	}
 }
 
+#define CORRELATE_ELEMENTS(_e1_, _e2_, kindstr)			\
+do {								\
+	typeof(_e1_) e1 = (_e1_);				\
+	typeof(_e2_) e2 = (_e2_);				\
+	e1->twin = e2;						\
+	e2->twin = e1;						\
+	/* set initial status, might change */			\
+	e1->status = e2->status = SAME;				\
+	if (strcmp(e1->name, e2->name)) {			\
+		/* Rename mangled element */			\
+		log_debug("renaming %s %s to %s\n",		\
+			  kindstr, e2->name, e1->name);		\
+		e2->name = strdup(e1->name);			\
+	}							\
+} while (0)
+
+#define UNCORRELATE_ELEMENT(_elem_)				\
+do {								\
+	typeof(_elem_) elem = (_elem_);				\
+	if (elem->twin->orig_name != elem->twin->name) {	\
+		free(elem->twin->name);				\
+		elem->twin->name = elem->twin->orig_name;	\
+	}							\
+	elem->twin->twin = NULL;				\
+	elem->twin = NULL;					\
+	if (elem->orig_name != elem->name) {			\
+		free(elem->name);				\
+		elem->name = elem->orig_name;			\
+	}							\
+} while (0)
+
+static void make_correlated_sections(struct section *sec1, struct section *sec2)
+{
+	CORRELATE_ELEMENTS(sec1, sec2, "section");
+}
+
+static void make_correlated_symbols(struct symbol *sym1, struct symbol *sym2)
+{
+	CORRELATE_ELEMENTS(sym1, sym2, "symbol");
+}
```

Comment:
Naming suggestions: make_correlated_sections() almost makes it sound like it's creating a section - how about renaming to kpatch_correlate_section()?   That also gives it a name analagous to its caller kpatch_correlate_sections().  And a similar comment for make_correlated_symbols().

And that would also imply a CORRELATE_ELEMENTS -> CORRELATE_ELEMENT rename for consistency - yes, it's correlating two elements, but depending on one's perspective they might be considered the same element :-)

Reply:
I was a bit worried about the potential kpatch_correlate_section/sections confusion.
But now that you mention it, I agree it makes more sense the way you suggest.
I'll do the upate.